zipline/finance/performance/tracker.py

```diff
@@ -396,6 +422,27 @@ def check_upcoming_dividends(self, midnight_of_date_that_just_ended):
             # notify periods to update their stats
             period.handle_dividends_paid(net_cash_payment)
 
+    def check_upcoming_asset_ends(self, next_trading_day):
```

Comment:
This method name doesn't make it obvious that this had an important and nontrivial side-effect.  I would name this something like `close_expired_asset_positions`.